src/lib/InstantSearch.ts

```diff
@@ -42,7 +55,25 @@ function defaultCreateURL() {
  * @fires Instantsearch#render This event is triggered each time a render is done
  */
 class InstantSearch extends EventEmitter {
-  constructor(options) {
+  public client: InstantSearchOptions['searchClient'];
+  public insightsClient: InsightsClient | null;
```

Comment:
```suggestion
  public insightsClient: InstantSearchOptions['insightsClient'] | null;
```

For the properties that come from the options we should be consistent (whatever solution we take).

Reply:
this is not the same, input is undefined or client, instance is client / null